src/ansible_navigator/configuration_subsystem/transform.py

```diff
@@ -7,13 +7,67 @@
 from typing import Dict
 from typing import List
 from typing import Tuple
+from typing import Union
 
 from ..utils.compatibility import importlib_resources
+from ..utils.functions import dict_merge_in_place_list_replace as dict_merge
+from ..utils.functions import shlex_join
 from .definitions import ApplicationConfiguration
 from .definitions import Constants
 from .definitions import SettingsEntry
+from .definitions import SettingsFileType
 from .defs_presentable import PresentableSettingsEntries
 from .defs_presentable import PresentableSettingsEntry
+from .utils import create_settings_file_sample
+
+
+def to_effective(
+    settings: ApplicationConfiguration,
+) -> SettingsFileType:
+    """Transform the current settings into a settings file.
+
+    :param settings: The current settings
+    :returns: The settings represented as settings file
+    """
+    rebuilt: Dict = {}
+    for entry in settings.entries:
+        path = entry.settings_file_path(prefix=settings.application_name_dashed)
+        if not isinstance(entry.value.current, Constants):
+            current: Union[bool, int, str, Dict, List] = entry.value.current
+            # It is necessary to un post-process here
+            if path == "ansible-navigator.ansible.cmdline":
+                # post-processed into a list
+                current = shlex_join(entry.value.current)
+            elif path == "ansible-navigator.execution-environment.volume-mounts":
+                current = []
+                for mount in entry.value.current:
+                    parts = mount.split(":")
+                    v_mount = {"src": parts[0], "dest": parts[1]}
```

Comment:
What if these indices don't exist? (Then the post processor would have failed at startup, but it feels awkward to rely on that here)

Reply:
After post-processing, they are all a string "src:dest:option":

https://github.com/ansible/ansible-navigator/blob/main/src/ansible_navigator/configuration_subsystem/navigator_post_processor.py#L387